### DISCUSSION TITLE
Worpress integration patch.js causing problems

### DISCUSSION THREAD

**Discussion by guggmann** [cables-gl/cables_docs#498] [Answers & Questions] [answered]
Hi there,

i wanted to use a cables patch for a wordpress site, but it seems the patch is causing some problems with other scripts after the patch.js is executed.
I did the integration with the cables wp plugin.
I have talked to our developer and he gave me this feedback:
The generated(?) patch.js file contains an invalid polyfill method String.prototype.endsWith which overrides the browsers native endsWith method:
`String.prototype.endsWith = function (e) {
  return this.match(e + "$") == e
}`
There are two problems with this method:

1. It overrides the native endsWith method that most browsers have: https://caniuse.com/?search=endswith 
Therefore, scripts executed later that use endsWith do not work properly either. 
The script should check if an endsWith method exists and override it only if it does not.
2. This method does not take into account that the parameter e might contain regular expression characters. In our particular case e is equal to ?, which is an invalid regular expression: "SyntaxError: nothing to repeat".

Also the method is overridden two times in the file.

my question now is, is there any chance to change this? because there is a conflict with our data privacy plugin.
hope someone can help.

thanks in advance

**Comment by pandrr** [comment, marked as answer]
thank you for reporting!

i changed it, so it does not get overwritten, if it already exists. please try to export from dev.cables.gl, there it is hopefully fixed.

**Reply by steam0r** [reply]
since the wordpress plugin can't export from dev you will have to do a manual export via the editor and replace patch.js...or wait for next release...

**Comment by guggmann** [comment]
ok, thanks for the fast answer. 
i will try it.
